### PR TITLE
M3-4556 Fix: long Domain records overflowing table rows

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

I had TXT records set to truncate at 255 characters, but a few users had records close to that count that were still breaking the display. I added truncation to every other field I could find where it was possible to break the display; if you can find others please let me know.

## Note to Reviewers

Make very long TXT, CAA, and SOA records. Have fun!

Screen Shot 2020-10-12 at 1 34 52 PM

### CODE REVIEW

**Review by johnwcallahan — CHANGES_REQUESTED** [review]
This is probably not realistic, but might as well address MX records too:

Screen Shot 2020-10-13 at 9 16 07 AM

More importantly, I'm easily able to break TXT record display with a medium-ish hostname:

Screen Shot 2020-10-13 at 9 14 22 AM

We should probably explicitly set the widths of these columns so we can precisely set the truncation points.

**Review by tiffwong — COMMENTED** [review]
I agree with @WilkinsKa1's suggestion, CSS might be the easiest way to handle this responsively

**Review by johnwcallahan — COMMENTED** [review]
Seems like we could get away with a larger maxWidth on higher breakpoints, still a lot of horizontal space:
Screen Shot 2020-10-15 at 7 43 29 AM

File: packages/manager/src/features/Domains/DomainRecords.tsx [hunk 1]
```diff
@@ -73,6 +72,12 @@ const styles = (theme: Theme) =>
       [theme.breakpoints.up('md')]: {
         maxWidth: 300,
         wordBreak: 'break-all'
```
Comment (WilkinsKa1) [inline packages/manager/src/features/Domains/DomainRecords.tsx:74]: I think we can remove this line

File: packages/manager/src/features/Domains/DomainRecords.tsx [hunk 2]
```diff
@@ -73,6 +72,12 @@ const styles = (theme: Theme) =>
...
+      },
+      '& .data': {
+        maxWidth: 300,
```
Comment (WilkinsKa1) [inline packages/manager/src/features/Domains/DomainRecords.tsx:77]: This is optional, but if we want to expand the amount of text displayed based on screensize, could add a breakpoint to increase this size for only large screens.

**Review by WilkinsKa1 — APPROVED** [review]
Looks good- left some optional feedback items, one for cleanup and one that could address the white space callout.

**Review by johnwcallahan — APPROVED** [review]
Any reason why `types.json` is updated here?

**Pull request merged**